chore: add translation context to strings

barredterra opened the pull request:
Add context for translateable strings

barredterra committed:
feat: add translation context

hasnain2808 commented on frappe/public/js/frappe/desk.js:
```diff
@@ -214,19 +214,20 @@ frappe.Application = class Application {
 					"fieldname": "submit",
 					"fieldtype": "Button",
-					"label": __("Submit")
+					"label": __("Submit", null, "Submit password for Email Account")
```
```suggestion
					"label": __("Submit", context="Submit password for Email Account")
```

hasnain2808 replied:
Thanks for taking out time to improve translations.
Let's use keyword argument instead of passing null
Let's do this for other places too

barredterra replied:
Hey @hasnain2808, [JavaScrip doesn't have named parameters](https://masteringjs.io/tutorials/fundamentals/parameters) and this is [in accordance with the docs](https://frappeframework.com/docs/v13/user/en/translations#6-adding-context-for-a-string).

hasnain2808 replied:
@barredterra thanks for pointing that out, mixed up python here.

hasnain2808 commented on frappe/public/js/frappe/list/list_view.js:
```diff
@@ -320,9 +320,9 @@ frappe.views.ListView = class ListView extends frappe.views.BaseList {
-				"Loading"
-			)}...</div>`
+			`<div class="freeze flex justify-center align-center text-muted">
+				${__("Loading...", null, "Freeze area in list view")}
```
are there different translations for Loading and Loading...
?

barredterra replied:
In general, translators have to know the whole string that will be displayed, not just an arbitrary part of it. But we can revert this change for now to not break existing translations.

barredterra committed:
revert: changes to translatable strings

This way existing translations will continue to work until the translation process is improved.

barredterra commented on frappe/public/js/frappe/list/list_view.js:
```diff
@@ -255,8 +255,13 @@ frappe.views.ListView = class ListView extends frappe.views.BaseList {
 
 	set_primary_action() {
 		if (this.can_create) {
+			const doctype_name = __(frappe.router.doctype_layout) || __(this.doctype)
+
+			// Better style would be __("Add {0}", [doctype_name], "Primary action in list view")
+			// Keeping it like this to not disrupt existing translations
+			const label = `${__("Add", null, "Primary action in list view")} ${doctype_name}`
```
```suggestion
			const doctype_name = __(frappe.router.doctype_layout) || __(this.doctype);

			// Better style would be __("Add {0}", [doctype_name], "Primary action in list view")
			// Keeping it like this to not disrupt existing translations
			const label = `${__("Add", null, "Primary action in list view")} ${doctype_name}`;
```

barredterra committed:
style: sider

barredterra commented:
@mergifyio backport version-13-hotfix
